tests/phpunit/tests/blocks/editor.php

```diff
@@ -568,6 +568,42 @@ public function test_block_editor_rest_api_preload_adds_missing_leading_slash( a
 		$this->assertStringContainsString( $expected, $haystack );
 	}
 
+	/**
+	 * @ticket 57547
+	 */
+	public function test_get_classic_theme_supports_block_editor_settings() {
+		$font_sizes = array(
+			array(
+				'name' => 'Small',
+				'size' => 12,
+				'slug' => 'small',
+			),
+			array(
+				'name' => 'Regular',
+				'size' => 16,
+				'slug' => 'regular',
+			),
+		);
+
+		add_theme_support( 'editor-font-sizes', $font_sizes );
+		$settings = get_classic_theme_supports_block_editor_settings();
+
+		$this->assertFalse( $settings['disableCustomColors'] );
+		$this->assertFalse( $settings['disableCustomFontSizes'] );
+		$this->assertFalse( $settings['disableCustomGradients'] );
+		$this->assertFalse( $settings['disableLayoutStyles'] );
+		$this->assertFalse( $settings['enableCustomLineHeight'] );
+		$this->assertFalse( $settings['enableCustomSpacing'] );
+		$this->assertFalse( $settings['enableCustomUnits'] );
+
+		$this->assertSame(
+			$font_sizes,
+			$settings['fontSizes']
+		);
+
+		remove_theme_support( 'editor-font-sizes' );
```

Comment:
This needs to be moved above the first assertion (before line 591), otherwise it will not be run if any assertion fails. That applies to any cleanup logic, it needs to happen before the assertions.
```suggestion
```